src/diskusage.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2015 Jolla Ltd.
+ * Contact: Thomas Perl <thomas.perl@jolla.com>
+ *
+ * You may use this file under the terms of the BSD license as follows:
+ *
+ * "Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Nemo Mobile nor the names of its contributors
+ *     may be used to endorse or promote products derived from this
+ *     software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE."
+ */
+
+#ifndef DISKUSAGE_H
+#define DISKUSAGE_H
+
+#include <QObject>
+#include <QVariant>
+#include <QJSValue>
+#include <QScopedPointer>
+
+class DiskUsagePrivate;
+
+class DiskUsage : public QObject
+{
+    Q_OBJECT
+    Q_DECLARE_PRIVATE(DiskUsage)
+
+    // True while calculation takes place
+    Q_PROPERTY(bool working
+            READ working
+            NOTIFY workingChanged)
+
+public:
+    explicit DiskUsage(QObject *parent=0);
+    virtual ~DiskUsage();
+
+    // Calculate the disk usage of the given paths, then call
+    // callback with a QVariantMap (mapping paths to usages in bytes)
+    Q_INVOKABLE void calculate(const QStringList &paths, QJSValue callback);
```

Comment:
The intended usage is for showing disk usage of Music, Videos, Documents, App Data, App Caches, Android Apps, etc... in the first step (that's already a big list, but it's mostly static, so could be done with one item per path).

In the second step it should allow to dig into the app data / app caches list and show per-app disk usage, which is potentially a big number of items - and that list is created by an "external" source (meaning that we'd need to at least glob directories with a certain pattern).

The list of glob patterns is probably static and small (app data, app config, app caches), but the list of paths these globs expand to might be big (for 50 installed apps it would be 150 paths -- one data/config/cache path for each app).